### PR TITLE
Site service1

### PR DESCRIPTION
**Pull request by naveed050**
Added site service endpoints

### CODE REVIEW

File: src/main/scala/com/imaginea/activegrid/core/models/SiteDeltaStatus.scala [hunk 1]
```diff
@@ -0,0 +1,27 @@
+package com.imaginea.activegrid.core.models
+
+/**
+  * Created by shareefn on 18/11/16.
+  */
+sealed trait SiteDeltaStatus {
+  def deltaStatus: String
+
+  override def toString: String = super.toString
```
Comment (eklavya) [inline src/main/scala/com/imaginea/activegrid/core/models/SiteDeltaStatus.scala:9]: why override here?

File: src/main/scala/com/imaginea/Main.scala [hunk 2]
```diff
@@ -1715,7 +1807,66 @@ object Main extends App {
...
+        (result._1 ::: instances, result._2 ::: reservedInstanceDetails, result._3 ::: loadBalancers, result._4 ::: scalingGroup)
+    }
+    val site1 = Site1(None, site.siteName, computedResult._1, computedResult._2, site.filters, computedResult._3, computedResult._4, List())
+    site1.toNeo4jGraph(site1)
+    site1
+  }
 
+  def saveCachedSite(siteId: Long): Boolean = {
+    val mayBeSite = cachedSite.get(siteId)
+    mayBeSite.exists { site =>
```
Comment (eklavya) [inline src/main/scala/com/imaginea/Main.scala]: Did you mean foreach?

File: src/main/scala/com/imaginea/activegrid/core/models/SiteDeltaStatus.scala (hunk 1 above)
Comment (naveed050) [inline src/main/scala/com/imaginea/activegrid/core/models/SiteDeltaStatus.scala]: yes, override shouldn't have been there. removing it now

File: src/main/scala/com/imaginea/Main.scala (hunk 2 above)
Comment (naveed050) [reply src/main/scala/com/imaginea/Main.scala]: I wanted  something like mayBeSite.map( return true).getOrElse(false)

File: src/main/scala/com/imaginea/activegrid/core/models/SiteDeltaStatus.scala (hunk 1 above)
Comment (naveed050) [reply src/main/scala/com/imaginea/activegrid/core/models/SiteDeltaStatus.scala:9]: we are using toString in custom json read and writes.. if we dont use ovverride, we get Method 'toString' needs override modifier

File: src/main/scala/com/imaginea/Main.scala (hunk 2 above)
Comment (eklavya) [reply src/main/scala/com/imaginea/Main.scala]: exists is shorter here. But that's not the purpose of exists.
Replace it with a foreach and mayBeSite.isDefined

Comment (naveed050) [reply src/main/scala/com/imaginea/Main.scala]: changed it Saurabh

**Pull request merged**